includes/class-wc-payments-apple-pay-registration.php

```diff
@@ -135,114 +129,11 @@ public function verify_domain_on_domain_name_change() {
 	 * Verify domain upon plugin update only in case the domain association file has changed.
 	 */
 	public function verify_domain_on_update() {
-		if ( $this->is_enabled() && ! $this->is_hosted_domain_association_file_up_to_date() ) {
```

Comment:
I was uncertain about this. Before the changes, we were performing the verification only if the domain association file was not "up to date".
With these changes, we always perform the verification on upgrade (if WooPayments is enabled and Google Pay/Apple Pay are enabled).
I don't think it's a major issue?

Reply:
Based on 1e99d3aeb9674869c3b05c9a1e35ceb5a3e0bd41, I think it used to be a large issue when for every plugin update, we got bombarded from the client site to the `apple_pay/domains` and `payment_method_domains` endpoints. 

I read the PR introducing https://github.com/Automattic/woocommerce-payments/pull/1239 this `on_update` action, its issue #1071, and the respective Stripe issues/PRs, and I think that's it's safe to remove this method completely. The current `verify_domain_on_domain_name_change` is already enough to detect the domain change, run the `register_domain` again.

Reply:
Thank you for locating that and for the background info @htdat ! I removed the hook and method in bbc3aea7b 👍